### PR TITLE
[cxxmodules] Fix tests for cxxmodules build on MacOSX

### PR DESCRIPTION
**Pull request by oshadura**
Fixes 2 tests for cxxmodules MacOSX builds http://cdash.cern.ch/testDetails.php?test=67555125&build=685983 && http://cdash.cern.ch/testDetails.php?test=67555125&build=685983

Originally test gives incorrect results also on non-cxxmodules build (ubuntu), but it is always green because expected output is test failure:
```
-- TEST COMMAND -- 
cd /mnt/build/night/LABEL/ROOT-ubuntu1804-clangHEAD/SPEC/cxxmod-noimt/V/6-18/build/roottest/cling/exception
/usr/bin/timeout -s USR2 270s root -l -b -q -e ''int*p=nullptr' *p'
-- BEGIN TEST OUTPUT --
-- END TEST OUTPUT --
-- BEGIN TEST ERROR --
Warning in <TApplication::GetOptions>: macro *p' not found
ROOT_cli_0:1:1: warning: missing terminating ' character [-Winvalid-pp-token]
'int*p=nullptr
^
ROOT_cli_0:1:1: error: expected expression

-- END TEST ERROR
```

If you will call directly from command line: `root -l -b -q -e "int*p=nullptr; *p"` you suppose to get next: https://gist.github.com/oshadura/1ee9cf4daf8cc4201d5ee63a164dbdda , giving exception about trying to dereference null pointer or trying to call routine taking non-null arguments.

Failure was noticeable in Mac OS X in cxxmodules builds, maybe due different way how bash works.

```
/build/jenkins/night/LABEL/mac1014/SPEC/cxxmod-noimt/build/roottest/cling/exception/root -l -b -q -e ''int*p=nullptr' *p'
-- BEGIN TEST OUTPUT --
-- END TEST OUTPUT --
CMake Error at /build/jenkins/night/LABEL/mac1014/SPEC/cxxmod-noimt/root/cmake/modules/RootTestDriver.cmake:190 (message):
  got exit code No such file or directory but expected 1
```
The original problem is in RootTestDriver.cmake, which removes all ";" and make some transformations on a COMMAND input string. Fixing/expanding RootTestDriver.cmake will take significant time (I already tried to fix it, but it takes longer time). So my proposal is to move from direct root invocation to use ROOT macro for this test.

Also in test directory, there is other macro used only for arm, which is testing the same functionality and for me it looks like it should be part of `if(arm).. else()`.

### CODE REVIEW

**Comment by oshadura** [comment]
@phsft-bot build!

File: cling/exception/CMakeLists.txt [hunk 1]
```diff
@@ -3,9 +3,9 @@ if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm")
...
                     PASSRC 1
                     LABELS roottest regression cling)
+else()
+  ROOTTEST_ADD_TEST(nullderef-e
+                    MACRO nullderef_exception.C
+                    PASSRC 1
+                    LABELS roottest regression cling)
 endif()
-
-ROOTTEST_ADD_TEST(nullderef-e
```
Comment (Axel-Naumann) [inline cling/exception/CMakeLists.txt]: I do not understand why this change is needed. The commit log does not explain either.

**Comment by oshadura** [comment]
@vgvassilev @Axel-Naumann I updated commit and description of this PR.

**Comment by Axel-Naumann** [comment]
Thanks, terrible :-( Did you try to simply escape (or double or triple escape) the `;`?

**Comment by oshadura** [comment]
@Axel-Naumann I suspect problem is that CMake is parcing command line string and just removing all semicolons: https://github.com/root-project/root/blob/master/cmake/modules/RootMacros.cmake#L1186 We can try to put it back after, but it will be even bigger damage.

**Comment by oshadura** [comment]
@Axel-Naumann can you take a look please?
